modules/core/src/arithm.cpp

```diff
@@ -947,7 +947,60 @@ void cv::copyTo(InputArray _src, OutputArray _dst, InputArray _mask)
 namespace cv
 {
 
-static BinaryFuncC* getMulTab()
+static void mul8u16uWrapper(const uchar* src1, size_t step1,
+                               const uchar* src2, size_t step2,
+                               uchar* dst, size_t step, int width, int height,
+                               void* usrdata)
+{
+    double scale = *((double*)usrdata);
+    cv_hal_mul8u16u(src1, step1, src2, step2, (ushort*)dst, step, width, height, scale);
+}
+
+static void mul8s16sWrapper(const uchar* src1, size_t step1,
+                               const uchar* src2, size_t step2,
+                               uchar* dst, size_t step, int width, int height,
+                               void* usrdata)
+{
+    double scale = *((double*)usrdata);
+    cv_hal_mul8s16s((schar*)src1, step1, (schar*)src2, step2, (short*)dst, step, width, height, scale);
+}
+
+static bool checkHalMulExtend()
+{
+    bool works = true;
+
+    // check that HAL functions are presented and work properly
+    uchar ua = 0, ub = 0;
+    ushort uc = 0;
+    int res;
+    res = cv_hal_mul8u16u(/* src1_data */ &ua, /* src1_step */ 1, /* src2_data */ &ub, /* src2_step */ 1,
+                             /* dst_data */ &uc, /* dst_step */ 1, /* width */ 1, /* height */ 1, /* scale */ 1);
+    if (res == CV_HAL_ERROR_NOT_IMPLEMENTED)
```

Comment:
The HAL function may return not implemented status, if size is to small or data is not aligned. Static pre-check does not work.